controllers/yawol-cloud-controller/targetcontroller/service_controller.go

```diff
@@ -57,14 +57,9 @@ func (r *ServiceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	className, ok := svc.Annotations[yawolv1beta1.ServiceClassName]
-	if !ok {
-		className = ""
-	}
-
 	infraDefaults := GetMergedInfrastructureDetails(r.InfrastructureDefaults, svc)
 
-	if className != r.ClassName {
+	if !helper.CheckLoadBalancerClasses(svc, r.ClassNames) {
```

Comment:
Nit: let's move this up before `infraDefaults := ...`. This should be the first thing to check during recocniliation.

Reply:
No because we delete LoadBalancers if a loadbalancer exists for a services with a "wrong" class. So it is possible to change the class.

Reply:
Nvm, that was a stupid comment from my side... 🙈